src/js/iov/Iov.js

```diff
@@ -351,16 +395,23 @@ export default class Iov extends EventEmitter {
       this.logger.error(`Error while creating / playing the player for stream ${this.streamConfiguration.streamName}`);
       this.logger.error(error);
 
+      this.destroyAllLoadingAnimations();
+
       throw error;
     }
 
     if (!iovPlayerId) {
+      this.destroyAllLoadingAnimations();
```

Comment:
When you say "move this check" - do you mean the check for the iovPlayerId?

I'm not sure that check should be moved inside the try or the catch. It also seems to be outside the scope of this ticket. Maybe I don't understand.

Reply:
@iwalston Thanks for pushing back on this. I was probably a bit too progressive in my attempts to optimize a few things here. Since the if statement `if (!iovPlayerId)` already existed prior to your changes it's fair to assume it's there for a reason for now (at least for the scope of this ticket).

For clarity, my thought was well line 384 (inside the try) would probably throw an error if it didn't return anything so I was questioning the need for the falsey check `!iovPlayerId`.